zerver/tests/test_events.py

```diff
@@ -694,20 +697,32 @@ def test_typing_events(self) -> None:
         check_typing_stop("events[0]", events[0])
 
     def test_custom_profile_fields_events(self) -> None:
+        realm = self.user_profile.realm
+        try_add_realm_custom_profile_field(
+            realm=realm, name="Expertise", field_type=CustomProfileField.LONG_TEXT
+        )
+
         events = self.verify_action(
-            lambda: notify_realm_custom_profile_fields(self.user_profile.realm, "add"),
+            lambda: notify_realm_custom_profile_fields(self.user_profile.realm),
             state_change_expected=False,
         )
         check_custom_profile_fields("events[0]", events[0])
 
-        realm = self.user_profile.realm
         field = realm.customprofilefield_set.get(realm=realm, name="Biography")
         name = field.name
         hint = "Biography of the user"
         try_update_realm_custom_profile_field(realm, field, name, hint=hint)
 
         events = self.verify_action(
-            lambda: notify_realm_custom_profile_fields(self.user_profile.realm, "add"),
+            lambda: notify_realm_custom_profile_fields(self.user_profile.realm),
+            state_change_expected=False,
+        )
```

Comment:
So, the `state_change_expected=False` here is definitely the result of a preexisting nasty bug in this test; I've pushed an additional commit that fixes this.  I encourage you to give it a read.

Reply:
I also noticed that states are not changed when adding, updating or removing profile fields, I thought there is something about the codebase that I'm not familiar with, but now that you've pointed out, it's clear that we were not actually verifying function but the notify functions. This is a mistake on my part, I should've reported. I'll be more careful next time.